packages/common/src/extensions/slickRowBasedEdit.ts

```diff
@@ -0,0 +1,596 @@
+      );
+    }
+  }
+
+  protected undoRowEdit(item: any) {
+    const idProperty = this._grid.getOptions().datasetIdPropertyName ?? 'id';
+    const targetRow = this._editedRows.get(item[idProperty]);
+    const row = this._grid.getData().getRowByItem(item);
+    if (
+      (row !== undefined &&
+        targetRow?.editCommands &&
+        targetRow.editCommands.length) ||
+      SlickGlobalEditorLock.cancelCurrentEdit()
+    ) {
+      while (targetRow!.editCommands.length > 0) {
+        const lastEdit = targetRow!.editCommands.pop();
+        if (lastEdit) {
+          lastEdit.undo();
+        }
+      }
+
+      targetRow!.columns.forEach((column) => {
+        this.removeUnsavedStylingFromCell(column, row!);
+      });
+      targetRow!.columns = [];
+
+      this._grid.invalidate();
+    }
+  }
+
+  protected renderUnsavedCellStyling(id: any, column: Column) {
+    if (column) {
+      const row = this._grid.getData()?.getRowById(id);
+      if (row !== undefined && row >= 0) {
+        const hash = { [row]: { [column.id]: ROW_BASED_EDIT_UNSAVED_CELL } };
+        const cssStyleKey = `${ROW_BASED_EDIT_UNSAVED_HIGHLIGHT_PREFIX}_${[
+          column.id,
+        ]}${row}`;
+        this._grid.setCellCssStyles(cssStyleKey, hash);
+        this._editedRows.get(id)?.cssStyleKeys.push(cssStyleKey);
+      }
+    }
+  }
+
+  protected handleAllRowRerender(
+    _e: SlickEvent,
+    _args: OnRowsOrCountChangedEventArgs
+  ) {
+    this._editedRows.forEach((editedRow, key) => {
+      editedRow.cssStyleKeys.forEach((cssStyleKey) => {
+        this._grid.removeCellCssStyles(cssStyleKey);
+      });
+      editedRow.cssStyleKeys = [];
+      editedRow.columns.forEach((column) => {
+        this.renderUnsavedCellStyling(key, column);
+      });
+    });
+  }
+
+  protected removeUnsavedStylingFromCell(column: Column, row: number) {
+    const cssStyleKey = `${ROW_BASED_EDIT_UNSAVED_HIGHLIGHT_PREFIX}_${[
+      column.id,
+    ]}${row}`;
+    this._grid.removeCellCssStyles(cssStyleKey);
+  }
+
+  protected removeUnsavedStylingFromRow(row: number) {
+    this._grid.getColumns().forEach((column) => {
+      this.removeUnsavedStylingFromCell(column, row);
+    });
+  }
+
+  protected optionsUpdatedHandler(_e: Event, args: OnSetOptionsEventArgs) {
+    this._addonOptions = {
+      ...this._defaults,
+      ...args.optionsAfter.rowBasedEditOptions,
+    } as RowBasedEditOptions;
+  }
+
+  protected async onCellClickHandler(event: Event, args: any) {
+    const dataContext = args.dataContext;
+    const target = event.target as HTMLElement;
+    const idProperty = this._grid.getOptions().datasetIdPropertyName ?? 'id';
+    const targetRow = this._editedRows.get(dataContext[idProperty]);
+    if (
+      (target.classList.contains(BTN_ACTION_DELETE) ||
+        target.parentElement?.classList.contains(BTN_ACTION_DELETE)) &&
+      this._gridService
+    ) {
+      if (this._addonOptions?.actionButtons?.deleteButtonPrompt) {
+        if (
```

Comment:
is it more readable to merge these 2 `if` into a single one? ... I guess it's ok as is since I see the next few conditions are much longer, so we might as well stay consistent